### ISSUE TITLE
"The Netherlands" doesn't match to NL country

### ISSUE THREAD

**Issue by brauliobo** [countries/countries#491]
Doing

    c = ISO3166::Country.find_country_by_name('The Netherlands')

doesn't match the country 'NL', just because of the `The`. Is there an easy fix to improve the search?

**Comment by rposborne** [comment]
Generally, `name` is only for developers I would target the translation tables if you are looking for a more.... "fuzzy" match.

**Comment by wvengen** [comment]
I'd guess that _The Netherlands_ fits in NL's `unofficial_names`, it isn't there yet.
Is there a more 'global' search? I guess it would be something like going through all entries and trying to match translations and unofficial names, right?

**Comment by wvengen** [comment]
I'm currently using:
```ruby
def find_code(name)
  c = ISO3166::Country.new(name)&.data
  c ||= ISO3166::Country.find_country_by_name(name)&.data
  c ||= ISO3166::Country.find_all_by('translated_names', name).values.first
  c ||= ISO3166::Country.find_all_by('unofficial_names', name).values.first
  c&.fetch('alpha2', nil)
end
```

**Comment by pmor** [comment]
This has been fixed in #614
